CognitoYourUserPools-Sample/Swift/CognitoYourUserPoolsSample/UserDetailTableViewController.swift

```diff
@@ -18,16 +18,23 @@
 import Foundation
 import AWSCognitoIdentityProvider
 
-class UserDetailTableViewController : UITableViewController {
+class UserDetailTableViewController : UITableViewController, AWSIdentityProviderManager {
     
     var response: AWSCognitoIdentityUserGetDetailsResponse?
     var user: AWSCognitoIdentityUser?
     var pool: AWSCognitoIdentityUserPool?
+    var awsCognitoCredentialsProvider: AWSCognitoCredentialsProvider?
```

Comment:
Small Swift-specific nitpick -- this field is declared as an optional (?). While it technically has to be optional, because you aren't assigning a value in the constructor, in practice it's not _really_ an optional. You can make the code feel nicer by either making this an implicitly unwrapped optional (i.e. a variable that is technically optional, but the rest of your code treats it as if it isn't):
`var awsCognitoCredentialsProvider: AWSCognitoCredentialsProvider!`

Another option maybe is to just assign it right away with a (lazy) initializer:
`lazy var awsCognitoCredentialsProvider = AWSCognitoCredentialsProvider(regionType: .USEast2, identityPoolId: CognitoIdentityPoolId)`